### PR TITLE
Feature/new dataset

### PR DESCRIPTION
**Pull request by Verntrax**

### CODE REVIEW

File: src/config.R [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
-credit_card_full = paste(resources_path, credit_card_file_full, sep = "")
-credit_card_10k = paste(resources_path, credit_card_file_10k, sep = "")
-credit_card_data = paste(resources_path, credit_card_file_final, sep = "")
+credit_card_file = "creditcard_final.csv"
+frogs_file = "Frogs_MFCCs.csv"
+
+credit_card_data = paste(resources_path, credit_card_file, sep = "")
+frog_data = paste(resources_path, frogs_file, sep = "")
+
+extract_normal_anomaly= function() {
```
Comment (unitatem) [inline src/config.R]: space=

File: src/main_kmeans.R [hunk 2]
```diff
@@ -0,0 +1,7 @@
+source("src/grouping_experiment.R")
+source("src/kmeans_wrapper.R")
+
+message("START")
+grouping_experiment(grouping_kmeans())
+message("END")
+rm(list = ls())
```
Comment (unitatem) [inline src/main_kmeans.R]: do it before not after

File: src/main_pam.R [hunk 3]
```diff
@@ -0,0 +1,7 @@
+source("src/grouping_experiment.R")
+source("src/pam_wrapper.R")
+
+message("START")
+grouping_experiment(grouping_pam())
+message("END")
+rm(list = ls())
```
Comment (unitatem) [inline src/main_pam.R]: do it before not after

File: src/grouping_experiment.R [hunk 4]
```diff
@@ -0,0 +1,21 @@
+source("src/config.R")
+source("src/grouping_wrapper.R")
+
+grouping_experiment = function(algorithm) {
+  normal_anomaly = extract_normal_anomaly()
```
Comment (unitatem) [inline src/grouping_experiment.R]: i dont like the idea of hardcoded function in config that has to be decoreted for proper alg functionality
maybe second arg in function for passing extract_normal_anomaly extraction function handler

File: src/main_kmeans.R [hunk 5]
```diff
@@ -0,0 +1,7 @@
+source("src/grouping_experiment.R")
+source("src/kmeans_wrapper.R")
+
+message("START")
+grouping_experiment(grouping_kmeans())
```
Comment (unitatem) [inline src/main_kmeans.R]: passing here extract_normal_anomaly fun handler looks not bad and solution seems to look a bit more general

File: src/main_pam.R [hunk 6]
```diff
@@ -0,0 +1,7 @@
+source("src/grouping_experiment.R")
+source("src/pam_wrapper.R")
+
+message("START")
+grouping_experiment(grouping_pam())
```
Comment (unitatem) [inline src/main_pam.R]: as above

File: src/grouping_wrapper.R [hunk 7]
```diff
@@ -38,10 +45,15 @@ anomaly_detector <- setRefClass("anomaly_detector",
...
+                                      else
+                                        dist_coeff[i] <<- quantile((dist_vec - distance[i]) / std_dev[i], 1 - tolerance)
                                     }
                                   },
                                   
-                                  predict = function(data, dist_coeff="numeric") {
+                                  predict = function(data) {
                                     data = sweep(data, 2, min)
                                     data = sweep(data, 2, max-min, "/")
                                     result = 0
```
Comment (unitatem) [inline src/grouping_wrapper.R]: preallocate

**Review by unitatem — CHANGES_REQUESTED** [review]

File: src/config.R (hunk 1 above)
Comment (Verntrax) [reply src/config.R]: Done

File: src/main_kmeans.R (hunk 2 above)
Comment (Verntrax) [reply src/main_kmeans.R]: Done

File: src/main_pam.R (hunk 3 above)
Comment (Verntrax) [reply src/main_pam.R]: Done

File: src/grouping_wrapper.R (hunk 7 above)
Comment (Verntrax) [reply src/grouping_wrapper.R]: Done

File: src/main_pam.R (hunk 6 above)
Comment (Verntrax) [reply src/main_pam.R]: Done

File: src/main_kmeans.R (hunk 5 above)
Comment (Verntrax) [reply src/main_kmeans.R]: Done

File: src/grouping_experiment.R (hunk 4 above)
Comment (Verntrax) [reply src/grouping_experiment.R]: Done

**Review by unitatem — APPROVED** [review]

**Pull request merged**